operator/pkg/client/factory.go

```diff
@@ -63,11 +65,19 @@ type ClientFactory interface {
 	// interface to properly initialize.
 	RedpandaAdminClient(ctx context.Context, object client.Object) (*rpadmin.AdminAPI, error)
 
+	// SchemaRegistryClient initializes an sr.Client based on the spec of the passed in struct.
+	// The struct *must* implement either the v1alpha2.SchemaRegistryConnectedObject interface of the v1alpha2.ClusterReferencingObject
```

Comment:
?
```suggestion
	// The struct *must* implement either the v1alpha2.SchemaRegistryConnectedObject interface or the v1alpha2.ClusterReferencingObject
```

What happens if `obj` doesn't implement either interface?

Reply:
same as the other initialization methods, we get a `ErrInvalidRedpandaClientObject` returned.